django_short_urls/auth.py

```diff
@@ -11,6 +11,20 @@
 from django.http import HttpResponse
 
 
+def _safe_base64_decode(encoded):
+    """Method that base64 decodes a string, while gracefully handing insufficient padding.
+    Returns unicode, assuming utf-8 charset"""
+    try:
+        # Add extra padding just in case, because python is very strict and it can't hurt
+        return (encoded + "===").decode('base_64').decode('utf8')
+    except Exception as err:  # pylint: disable=broad-except
+        if err.message == 'Incorrect padding':
+            # Adding the extra padding failed, there's no way we'll decode this
```

Comment:
Agreed. In which case, empty logins are hardly arguable.
Should we do it within this PR? Looks like it's just a matter of adding [`required = True` to the mongoengine fields](https://github.com/MongoEngine/mongoengine/blob/v0.9.0/mongoengine/base/fields.py#L384-L386)